### ISSUE TITLE
Datastore query not returning error when network is disconnected

### ISSUE THREAD

**Issue by mcfarljw** [GoogleCloudPlatform/google-cloud-node#2137]
#### Environment details

  - OS: macOS 10.12.3
  - Node.js version: 6.10.0
  - npm version: 3.10.10
  - google-cloud-node version: 0.7.1 (datastore)

#### Steps to reproduce

  1. require `@google-cloud/datastore": "~0.7"`
  2. make a basic recursive call using a query and a cursor
  
```js
function run (endCursor) {
	var query = datastore.createQuery('Animal');

	query.start(endCursor);
	query.limit(300);

	query.run(
		function (error, entities, info) {
			if (error) {
				console.log(error);
			}

			console.log(entities);

			if (info.endCursor) {
				run(info.endCursor);
			}
		}
	);
}
```

  3. disconnect the network while running the function
  4. wait a few minutes and notice no network error is returned

**Comment by stephenplusplus** [comment]
Thanks for reporting. I reproduced, and I believe *part* of this is our retry strategy at work. Because if you re-connect, it picks up where it left off.

I am a bit confused on some behavior coming from our transport layer, `grpc`, so I've opened an issue there: https://github.com/grpc/grpc/issues/10322 -- this may turn out to relate. The gist of it is, without a network connection, the process stays open until the network is re-connected. I'm not sure if this is a bug or intentional retry behavior.
